frontend-api/src/test/java/uk/gov/di/authentication/frontendapi/lambda/MfaHandlerTest.java

```diff
@@ -380,9 +383,22 @@ void shouldReturn400IfUserHasReachedTheSmsSignInCodeRequestLimit(JourneyType jou
         assertThat(result, hasJsonBody(ErrorResponse.ERROR_1025));
 
         var codeRequestType = CodeRequestType.getCodeRequestType(MFAMethodType.SMS, journeyType);
-        verify(codeStorageService)
-                .saveBlockedForEmail(
-                        EMAIL, CODE_REQUEST_BLOCKED_KEY_PREFIX + codeRequestType, LOCKOUT_DURATION);
+
+        if (journeyType == JourneyType.REAUTHENTICATION) {
```

Comment:
Suggestion:

```        
if (journeyType != JourneyType.REAUTHENTICATION || !reauthEnabled) {
                verify(codeStorageService)
                        .saveBlockedForEmail(
                                EMAIL,
                                CODE_REQUEST_BLOCKED_KEY_PREFIX + codeRequestType,
                                LOCKOUT_DURATION);
        } else {
            verifyNoInteractions(codeStorageService);
        }

verify(sessionService).save(argThat((session) ->
                session.getCodeRequestCount(NotificationType.MFA_SMS, journeyType) == 0));
```

(I found the nested conditions a bit confusing, and would be good to verify what happens on a reauth journey with reauth enabled, and to clarify that we wipe the count in any case)

Reply:
Added some clarifying methods to make this clearer.

Reply:
Cool LGTM, I think it's still worth that extra assertion ```verify(sessionService).save(argThat((session) ->
                session.getCodeRequestCount(NotificationType.MFA_SMS, journeyType) == 0))``` being added in